fishtank/src/cluster.ts

```diff
@@ -33,14 +37,38 @@ export class Cluster {
     })
   }
 
-  async spawn(options: { name: string; image?: string }): Promise<Node> {
+  async spawn(options: {
+    name: string
+    image?: string
+    config?: Partial<ConfigOptions>
+  }): Promise<Node> {
     const containerName = this.containerName(options.name)
-    await this.backend.runDetached(options.image ?? DEFAULT_IMAGE, {
+
+    const runOptions: RunOptions = {
       name: containerName,
       networks: [this.networkName()],
       hostname: options.name,
       labels: { [CLUSTER_LABEL]: this.name },
-    })
+    }
+
+    if (options.config) {
+      const args: string[] = []
+
+      const configString = JSON.stringify(options.config)
+
+      const dest = join(tmpdir(), 'fishtank', containerName, '.ironfish')
+      await promises.mkdir(dest, {
+        recursive: true,
+      })
+
+      await promises.writeFile(resolve(dest, 'config.json'), configString)
+
+      args.push('--datadir=/fishtank/.ironfish')
+      runOptions.volumes = new Map<string, string>([[dest, '/fishtank/.ironfish']])
```

Comment:
Like before, if you use:

```
[dest, '/fishtank/.ironfish']
```

instead of:

```
[dest, '/root/.ironfish']
```

then there's no need for `--datadir`

Reply:
root folder has different permission, it does not work without modification of the permission. I would prefer a different folder so no modification is needed.

Reply:
The ironfish process should run as root (in its uid namespace) inside the container. There shouldn't be any permissions problems. What error are you seeing exactly?

Reply:
```
2023-09-18 14:45:28     Error: EPERM: operation not permitted, open 
2023-09-18 14:45:28     '/root/.ironfish/internal.json'
2023-09-18 14:45:28     Code: EPERM
```

Reply:
Mh, this must be a macOS-specific behavior. Looking on the web (if I understood correctly), it looks like you can end up in that state if you get a prompt from macOS and you deny Docker access to a certain folder. For example here: https://stackoverflow.com/q/58482352 and https://stackoverflow.com/q/47382957. Is it possible that this has happened? Otherwise I do not see why you would get an error for `/root`, but not for `/ironfish`. Both should end up being owned by the same user that runs the init process in the container. Also note that you're getting an EPERM, not an EACCES, which is a clear symptom that this is not a permission problem, but more the kernel forbidding you to access certain files. Some posts suggest that reinstalling Docker may make the issue go away.